clients/src/main/java/org/apache/kafka/clients/admin/SharePartitionOffsetInfo.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.clients.admin;
+
+import org.apache.kafka.common.annotation.InterfaceStability;
+
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * This class is used to contain the offset and lag information for a share-partition.
+ */
+@InterfaceStability.Evolving
+public class SharePartitionOffsetInfo {
+    private final long startOffset;
+    private final Optional<Integer> leaderEpoch;
+    private final Optional<Long> lag;
+
+    /**
+     * Construct a new SharePartitionOffsetInfo.
+     *
+     * @param startOffset The share-partition start offset
```

Comment:
Sorry, I'm a bit confused by this "start offset". For example, If a partition has a latest offset of `23547540` and a share consumer has a "start offset" of `23541472`, how do we explain the difference between those numbers?

```
chia7712@fedora:~/project/kafka$ ./bin/kafka-get-offsets.sh --bootstrap-server localhost:20000 --topic chia
chia:0:23547540
chia7712@fedora:~/project/kafka$ ./bin/kafka-share-groups.sh  --bootstrap-server localhost:20000 --offsets --describe --all-groups

GROUP               TOPIC           PARTITION  START-OFFSET  LAG
perf-share-consumer chia            0          23541472      -
```

Reply:
@AndrewJSchofield thanks for your response! I have a follow-up question.

> The start offset is a little like the committed offset for a consumer group. 

Will the start offset eventually equal to the latest offset of the partition?

Reply:
> we are expressing progress delivering records and a measure of the records whose delivery has not yet completed.

Pardon me, the start offset is 0 even though the offset is consumed and committed by share consumer? Is it correct?

Reply:
The update of the start offset is lazy for performance reasons. Sometimes, it's very slightly behind where you might expect, but next time we write the start offset to the `__share_group_state` topic, it moves forwards. As long as the delivery state of the record is accurate, having the SPSO a little behind doesn't matter from a logical standpoint.